code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -2891,3 +2891,29 @@
 
 	if(SPT_PROB(10, seconds_per_tick))
 		carbon_metabolizer.set_heartattack(TRUE)
+
+/datum/reagent/hauntium
+	name = "Hauntium"
+	color = "#3B3B3BA3"
+	description = "An eerie liquid created by purifying the prescence of ghosts. If it happens to get in your body, it starts hurting your soul." //soul as in mood and heart
+	taste_description = "evil spirits"
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	material = /datum/material/hauntium
+	ph = 10
+	var/time_multiplier = 15 SECONDS
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/reagent/hauntium/expose_obj(obj/exposed_obj, volume) //gives 15 seconds of haunting effect for every unit of it that touches an object
+	. = ..()
+	exposed_obj.AddElement(/datum/element/haunted, 0)
+	addtimer(CALLBACK(exposed_obj, PROC_REF(_RemoveElement), list(/datum/element/haunted, 0)), volume * time_multiplier, TIMER_UNIQUE|TIMER_OVERRIDE)
```

Comment:
This feels a bit icky. Beside the callback's proc having an underscore prefix, which usually means it shouldn't be deliberately called without using a wrapper of some sort, this is likely going to conflict with the hauntium material and revenant ability, which also add and remove the same element, at some point.

One solution to the problem would be adding an `atom/movable` proc that checks if the movable already has a haunted trait, add the element if that statement is false, then adds the trait. As well as  another that does the opposite.

More or less like this:
```
/atom/movable/proc/make_haunted(source, color)
	if(!HAS_TRAIT(src, TRAIT_HAUNTED))
		AddElement(/datum/element/haunted, color)
	ADD_TRAIT(src, TRAIT_HAUNTED, source)

/atom/movable/proc/remove_haunted(source)
	REMOVE_TRAIT(src, TRAIT_HAUNTED, source)
	if(!HAS_TRAIT(src, TRAIT_HAUNTED))
		RemoveElement(/datum/element/haunted)
```

Reply:
I just took it from the gravitums code but instead of no gravity it gives it haunting, I can try and do something more like that though

Reply:
if I were to put those procs in the haunted element page, datums/elements/haunted.dm, how would I call them from reagents / anywhere else I may want to use it?

Reply:
As long as it's an atom/movable proc, you can basically call `exposed_obj.make_haunted(trait_source_define_goes_here, color_string_goes_here)` and `exposed_obj.remove_haunted(trait_source_define_goes_here)` and you should be set.